### PR TITLE
feat(workers): implement worker service add/set/remove api tags

### PR DESCRIPTION
**Pull request by A-440Hz**

### CODE REVIEW

File: internal/proto/controller/api/services/v1/worker_service.proto [hunk 1]
```diff
@@ -126,3 +167,39 @@ message DeleteWorkerRequest {
...
+message AddWorkerTagsRequest {
+  string id = 1; // @gotags: `class:"public"`
+  // Version is used to ensure this resource has not changed.
+  // The mutation will fail if the version does not match the latest known good version.
+  uint32 version = 2; // @gotags: `class:"public"`
+  repeated servers.v1.TagPair tags = 3 [json_name = "tags"]; // @gotags: `class:"public"`
+}
+
+message AddWorkerTagsResponse {
+  resources.workers.v1.Worker item = 1;
```
[the remark this answers is not in the document]
Comment (A-440Hz) [reply internal/proto/controller/api/services/v1/worker_service.proto:180]: I'm not certain what you mean. Should it return all api_tags or just the ones which were added in the request? Should I do the same with SetWorkerTagsResponse?

[the remark this answers is not in the document]
Comment (A-440Hz) [reply internal/proto/controller/api/services/v1/worker_service.proto:180]: thanks!

File: internal/proto/controller/api/services/v1/worker_service.proto [hunk 2]
```diff
@@ -126,3 +167,39 @@ message DeleteWorkerRequest {
 }
 
 message DeleteWorkerResponse {}
+
+message AddWorkerTagsRequest {
+  string id = 1; // @gotags: `class:"public"`
+  // Version is used to ensure this resource has not changed.
+  // The mutation will fail if the version does not match the latest known good version.
+  uint32 version = 2; // @gotags: `class:"public"`
+  repeated servers.v1.TagPair tags = 3 [json_name = "tags"]; // @gotags: `class:"public"`
```
[the remark this answers is not in the document]
Comment (A-440Hz) [reply internal/proto/controller/api/services/v1/worker_service.proto:176]: We can't have duplicate keys with this change. (Or at least I can't create a struct in my test with many duplicate keys at once). Will this be a problem beyond testing?

**Comment by A-440Hz** [comment]
make api is generating my `worker.gen.go` weird even though my proto field looks the same as config_tags and canonical_tags. Do you know why this could be the case?

```
 type Worker struct {
	Id                       string              `json:"id,omitempty"`
	ScopeId                  string              `json:"scope_id,omitempty"`
	Scope                    *scopes.ScopeInfo   `json:"scope,omitempty"`
	Name                     string              `json:"name,omitempty"`
	Description              string              `json:"description,omitempty"`
	CreatedTime              time.Time           `json:"created_time,omitempty"`
	UpdatedTime              time.Time           `json:"updated_time,omitempty"`
	Version                  uint32              `json:"version,omitempty"`
	Address                  string              `json:"address,omitempty"`
	CanonicalTags            map[string][]string `json:"canonical_tags,omitempty"`
	ConfigTags               map[string][]string `json:"config_tags,omitempty"`
	LastStatusTime           time.Time           `json:"last_status_time,omitempty"`
	WorkerGeneratedAuthToken string              `json:"worker_generated_auth_token,omitempty"`
	ActiveConnectionCount    uint32              `json:"active_connection_count,omitempty"`
	Type                     string              `json:"type,omitempty"`
	ApiTags                  []*v1.ApiTagsEntry  `json:"api_tags,omitempty"`
	AuthorizedActions        []string            `json:"authorized_actions,omitempty"`

	response *api.Response
}
```

**Comment by irenarindos** [comment]
> 

I think you need to update parse.go and that should take care of it: https://github.com/hashicorp/boundary/blob/main/internal/api/genapi/parse.go#L73-L74

**Comment by A-440Hz** [comment]
thanks! that fixed it

**Pull request merged**